File: output/ChatGPT/avatar/Java/C++/codeforces_131_B.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;
    const int N = 1e5 + 5;
    vector<long long> cnt(21, 0);
    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        cnt[num + 10]++;
    }
    long long res = 0;
    for (int i = 0; i < 10; i++) {
        res += cnt[i] * cnt[20 - i];
    }
    res += (cnt[10] * (cnt[10] - 1)) / 2;
    cout << res << endl;

    return 0;
}

File: output/ChatGPT/avatar/Java/Python/codeforces_131_B.py
n = int(input())
cnt = [0] * 21

for i in range(n):
    cnt[int(input()) + 10] += 1

res = 0
for i in range(10):
    res += cnt[i] * cnt[20 - i]

res += (cnt[10] * (cnt[10] - 1)) // 2

print(res)